src/category_theory/groupoid/subgroupoid.lean

```diff
@@ -329,23 +436,91 @@ def map (hφ : function.injective φ.obj) (S : subgroupoid C) : subgroupoid D :=
   end,
   mul := begin
     rintro _ _ _ _ ⟨f,hf⟩ q hq,
-    obtain ⟨c₃,c₄,g,he,rfl,hg,gq⟩ := (map.mem_arrows_iff φ hφ S q).mp hq,
+    obtain ⟨c₃,c₄,g,he,rfl,hg,gq⟩ := (map.arrows_iff φ hφ S q).mp hq,
     cases hφ he, rw [gq, ← eq_conj_eq_to_hom, ← φ.map_comp],
     split, exact S.mul hf hg,
   end }
 
+lemma mem_map_iff (hφ : function.injective φ.obj) (S : subgroupoid C) {c d : D} (f : c ⟶ d):
+  f ∈ (map φ hφ S).arrows c d ↔
+  ∃ (a b : C) (g : a ⟶ b) (ha : φ.obj a = c) (hb : φ.obj b = d) (hg : g ∈ S.arrows a b),
+    f = (eq_to_hom ha.symm) ≫ φ.map g ≫ (eq_to_hom hb) := map.arrows_iff φ hφ S f
+
+lemma galois_connection_map_comap (hφ : function.injective φ.obj) :
+  galois_connection (map φ hφ) (comap φ) :=
+begin
+  rintro S T, simp_rw [le_iff], split,
+  { exact λ h c d f fS, h (map.arrows.im f fS), },
+  { rintros h _ _ g ⟨a,gφS⟩,
+    exact h gφS, },
+end
+
 lemma map_mono (hφ : function.injective φ.obj) (S T : subgroupoid C) :
   S ≤ T → map φ hφ S ≤ map φ hφ T :=
 by { rintro ST ⟨c,d,f⟩ ⟨_,h⟩, split, exact @ST ⟨_,_,_⟩ h }
```

Comment:
This could be proved by the Galois connection. 
Also, there are several other Galois connection lemma that should also be ported, for example `comap_map_le` or `map_le_iff_le_comap`.